app/controllers/async_info_controller.rb

```diff
@@ -62,6 +63,10 @@ def user_data
     end.to_json
   end
 
+  def check_user_role
```

Comment:
Maybe it's just me, but I don't find the name of this method very intent-revealing.